File: backend/app/schemas/organization.py
from typing import Optional, List, Dict
from pydantic import BaseModel
from datetime import datetime


class OrganizationBase(BaseModel):
    """Base organization schema"""
    name: str
    description: Optional[str] = None
    # New fields from revised schema
    external_id: Optional[str] = None
    status: str = 'active'
    billing_email: Optional[str] = None
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    timezone: str = 'UTC'
    settings: Dict = {}


class OrganizationCreate(OrganizationBase):
    """Schema for creating organizations"""
    pass


class OrganizationUpdate(BaseModel):
    """Schema for updating organizations"""
    name: Optional[str] = None
    description: Optional[str] = None
    external_id: Optional[str] = None
    status: Optional[str] = None
    billing_email: Optional[str] = None
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    timezone: Optional[str] = None
    settings: Optional[Dict] = None


class OrganizationInDBBase(OrganizationBase):
    """Base schema for organizations in DB"""
    id: int
    created_at: datetime
    updated_at: datetime
    stripe_customer_id: Optional[str] = None
    
    class Config:
        from_attributes = True  # Updated from orm_mode = True for Pydantic v2 compatibility


class Organization(OrganizationInDBBase):
    """Schema for organization responses"""
    pass


class OrganizationWithStats(Organization):
    """Organization with usage statistics"""
    agent_count: int
    active_agent_count: int
    monthly_cost: float
    monthly_revenue: float

File: frontend/types/organization.ts
export interface Organization {
  id: number;
  name: string;
  description?: string;
  external_id?: string;
  status: string;
  billing_email?: string;
  contact_name?: string;
  contact_phone?: string;
  timezone: string;
  settings: Record<string, any>;
  created_at: string;
  updated_at: string;
  stripe_customer_id?: string;
}

export interface OrganizationWithStats extends Organization {
  agent_count: number;
  active_agent_count: number;
  monthly_cost: number;
  monthly_revenue: number;
}

export interface OrganizationCreate {
  name: string;
  description?: string;
  external_id?: string;
  status?: string;
  billing_email?: string;
  contact_name?: string;
  contact_phone?: string;
  timezone?: string;
  settings?: Record<string, any>;
}

export interface OrganizationUpdate {
  name?: string;
  description?: string;
  external_id?: string;
  status?: string;
  billing_email?: string;
  contact_name?: string;
  contact_phone?: string;
  timezone?: string;
  settings?: Record<string, any>;
}
